power_grid_model_c/power_grid_model/include/power_grid_model/math_solver/y_bus.hpp

```diff
@@ -317,6 +317,8 @@ template <bool sym> class YBus {
     }
     std::shared_ptr<IdxVector const> shared_diag_lu() const { return {y_bus_struct_, &y_bus_struct_->diag_lu}; }
 
+    constexpr auto& get_y_bus_struct() const { return y_bus_struct_; }
```

Comment:
to avoid confusion, let's more explicit

```suggestion
    constexpr auto& get_y_bus_structure() const { return y_bus_struct_; }
```